client/auth/controller.js

```diff
@@ -20,16 +22,27 @@ import userFactory from 'lib/user';
 import Main from 'components/main';
 import PulsingDot from 'components/pulsing-dot';
 
-module.exports = {
+export default {
 
-	// Login screen used by the desktop application
-	login: function() {
-		if ( OAuthToken.getToken() ) {
-			page( '/' );
-		} else {
-			ReactDom.render(
-				React.createElement( LoginComponent, {} ),
-				document.getElementById( 'primary' )
+	login: function( context ) {
+		if ( config.isEnabled( 'oauth' ) ) {
+			// hack to keep the blue background for Desktop login
+			document.querySelector( '.layout' ).style.backgroundColor = '#0087be';
```

Comment:
How important is it that both login pages share a route? If I recall correctly, the address bar isn't displayed in the desktop app, I think we could:

- Mount it on e.g. `/oauth-login` or something.
- Create separate sections for browser auth and desktop auth. This makes sense from a performance point of view, as we don't need to load `OAuthLogin` when oauth is disabled, and the same for `NativeLogin` when `native-login` is disabled.